### DISCUSSION TITLE
WiFi power save mode for ICYStream

### DISCUSSION THREAD

**Discussion by sneiss** [pschatzmann/arduino-audio-tools#1534] [Q&A] [answered]
If both, WiFi and Bluetooth are enabled, WiFi modem sleep has to be enabled. If not enabled, I get the following runtime error:
`esp32 Error! Should enable WiFi modem sleep when both WiFi and Bluetooth are enabled!!!!!!`
Activating power save mode when using an URLStream solves the problem. 

ICYStream is currently missing a function to enable power save mode. I added a simple function to the ICYStream class:
`void setPowerSave(bool ps) { url.setPowerSave(ps); }`

**Comment by pschatzmann** [comment, marked as answer]
The [method as been added.](https://github.com/pschatzmann/arduino-audio-tools/commit/3e4e24fed2f8805879caa72d86e3f7fe19b979d6)
